src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -102,12 +131,17 @@ private enum IndirectReturnState : byte
 
             _boundBody = Optimizer.Optimize(
                 boundBody, 
-                debugFriendly: _optimizations != OptimizationLevel.Release, 
+                debugFriendly: _ilEmitStyle != ILEmitStyle.Release, 
                 stackLocals: out _stackLocals);
 
             _methodBodySyntaxOpt = (method as SourceMethodSymbol)?.BodySyntax;
         }
 
+        private bool IsDebugPlus()
+        {
+            return this._module.Compilation.Options.DebugPlusMode;
```

Comment:
OptimizationLevel is a public API. This change specifically minimizes impact on public contracts. 
We can expose this feature via public APIs and use OptimizationLevel, if there is a strong enough scenario for that.